src/data_structure/ziplist/ziplist.c

```diff
@@ -353,12 +389,11 @@ ziplist_remove_val(uint32_t *removed, ziplist_p zl, const struct blob *val,
         return ZIPLIST_EINVALID;
     }
 
-    if (count == 0) {
-        return ZIPLIST_EINVALID;
-    }
-
+    count = count == 0 ? INT64_MAX : count;
```

Comment:
I feel this is probably me not explaining what I was thinking when implementing this function- I was leaning toward leaving the translation ("0 -> all") to the upper layer, similar to how we handled TTL timestamps in the protocol/storage, to keep the lower levels more straightforward. How do you feel about this?

Reply:
yea, i actually like the original behavior better too. the reason i changes this is that count == 0 for lrem in redis means remove all. i will revert this; perhaps we can move this logic out to protocol if we want to support the redis behavior.

Reply:
however, returning error also seems wrong to me. intuitively, it seems like this case should just be a no op

Reply:
Agree that noop is the better behavior.

This file is part of "data_structure", we can still put the translation in Redis' lrem implementation, e.g. in `src/server/redis/data/cmd_list.c` or somewhere like that.